Snowflake.Data/Core/HttpUtil.cs

```diff
@@ -404,8 +411,19 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                         }
                         else
                         {
-                            //TODO: Should probably check to see if the error is recoverable or transient.
-                            logger.Warn("Error occurred during request, retrying...", e);
+                            Exception innermostException = e;
+                            while (innermostException.InnerException != null) innermostException = innermostException.InnerException;
+
+                            if (innermostException is AuthenticationException)

```

Comment:
how is that compared to jdbc driver? Are there in jdbc driver any kinds of exceptions not retryable as well?

Reply:
JDBC does not retry if the exception is due to a [revoked certificate](https://github.com/snowflakedb/snowflake-jdbc/blob/cc073f8917b91dd0855985c3085b99c73fe066d5/src/main/java/net/snowflake/client/jdbc/RestRequest.java#L324)